### ISSUE TITLE
Kube-proxy not working on coreos/gce

### ISSUE THREAD

**Issue by yifan-gu** [kubernetes/kubernetes#15676]
After I created a service, the kube-proxy on other machines cannot establish connection to the pod in the service:
```
E1014 23:27:41.282669     870 proxysocket.go:104] Dial failed: dial tcp 172.17.0.5:80: i/o timeout
E1014 23:27:43.282881     870 proxysocket.go:104] Dial failed: dial tcp 172.17.0.5:80: i/o timeout
E1014 23:27:43.287773     870 proxysocket.go:104] Dial failed: dial tcp 172.17.0.5:80: no route to host
E1014 23:27:46.293828     870 proxysocket.go:104] Dial failed: dial tcp 172.17.0.5:80: no route to host
E1014 23:27:46.293858     870 proxysocket.go:138] Failed to connect to balancer: failed to connect to an endpoint.
```

The kube-proxy on the same node with the pod works fine:

**Comment by yifan-gu** [comment]
Think I need to fix this first before #15591

**Comment by yifan-gu** [comment]
cc @dchen1107

**Comment by dchen1107** [comment]
cc/ @runningwild who recently fix some issues for this setup.

cc/ @ArtfulCoder for kube-proxy related configuration

**Comment by thockin** [comment]
172 suggests the default docker networking - are you sure the bridge is
being configured with the per-node podCIDR?

**Comment by yifan-gu** [comment]
I think the problem is kubelet is not started with `--configure-cbr0=true`

**Comment by yifan-gu** [comment]
IIRC, the configuration is for rkt at that time, so the cbr0 is not set. But I am going to add this so we can make coreos/gce works for docker as well.

**Comment by dchen1107** [comment]
--configure-cbr0=true and related changes in kubelet was introduced after my initial setup for rkt. We need to upgrade this.

**Comment by yifan-gu** [comment]
@dchen1107 I made some changes to the config files, testing this.

**Comment by yifan-gu** [comment]
Pushed a fix for this:  #15749
